roc/e2e_test.go

```diff
@@ -86,78 +103,181 @@ func (e *e2e) close(t *testing.T) {
 }
 
 func TestEnd2End_Default(t *testing.T) {
-	e := newE2E(t)
-	defer e.close(t)
-
-	samplesCnt := 100
-	testSamples := make([]float32, samplesCnt)
-	for i := 0; i < samplesCnt/NumChannels; i++ {
-		testSamples[i*NumChannels] = float32(i+1) / 100
-		testSamples[i*NumChannels+1] = -float32(i+1) / 100
+	tests := []struct {
+		name   string
+		params e2eParams
+	}{
+		{
+			name: "default",
+			params: e2eParams{
+				clockSource: ClockExternal,
+				fecEncoding: FecEncodingDisable,
+				sourceURI:   "rtp://127.0.0.1:0"},
+		},
+		{
+			name: "fec",
+			params: e2eParams{
+				clockSource: ClockExternal,
+				fecEncoding: FecEncodingRs8m,
+				sourceURI:   "rtp+rs8m://127.0.0.1:0",
+				repairURI:   "rs8m://127.0.0.1:0"},
+		},
 	}
 
-	var interval = time.Second / time.Duration(44100/samplesCnt)
-	sendTicker := time.NewTicker(interval)
-	defer sendTicker.Stop()
-
-	receiveTicker := time.NewTicker(interval)
-	defer receiveTicker.Stop()
-
-	endChan := make(chan struct{})
-	var wait sync.WaitGroup
-	wait.Add(1)
-	go func() {
-		for {
-			select {
-			case <-sendTicker.C:
-				err := e.Sender.WriteFloats(testSamples)
-				if err != nil {
-					t.Fail()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := newE2E(t, tt.params)
+			defer e.close(t)
+
+			samplesCnt := 100
+			testSamples := generateTestSamples(samplesCnt)
+
+			var interval = time.Second / time.Duration(44100/samplesCnt)
```

Comment:
nit: Let's use sample rate from config, instead of 44100?

Reply:
done!